packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -774,7 +774,7 @@ class PubspecYaml {
               // First, make sure it's a unique dependency. Listing dependencies
               // twice doesn't make sense.
               if (masterDependencies.containsKey(dependency.name)) {
-                throw '$filename contains two dependencies on ${dependency.name}.';
+                throw Exception('$filename contains two dependencies on ${dependency.name}.');
```

Comment:
If this happens in a file under user control, it should also be a `throwToolExit`.